src/main/java/org/junit/internal/StackTraces.java

```diff
@@ -0,0 +1,203 @@
+package org.junit.internal;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.AbstractList;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Utility class for working with stack traces.
+ */
+public class StackTraces {
+    private StackTraces() {
+    }
+
+    /**
+     * Gets a trimmed version of the stack trace of the given exception. Stack trace
+     * elements that are below the test method are filtered out.
+     *
+     * @return a trimmed stack trace, or the original trace if trimming wasn't possible
+     */
+    public static String getTrimmedStackTrace(Throwable exception) {
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+        exception.printStackTrace(writer);
+        return trimStackTrace(exception.toString(), stringWriter.toString());
+    }
+
+    /**
+     * Trims the given stack trace.  Stack trace elements that are below the test method are
+     * filtered. out.
+     *
+     * @param fullTrace the full stack trace
+     * @return a trimmed stack trace, or the original trace if trimming wasn't possible
+     */
+    public static String trimStackTrace(String fullTrace) {
+        return trimStackTrace("", fullTrace);
+    }
+
+    static String trimStackTrace(String extracedExceptionMessage, String fullTrace) {
```

Comment:
@ashleyfrieze I'm a fan of the Hub tool for tasks like this (see http://blog.spreedly.com/2014/06/24/merge-pull-request-considered-harmful/#). That being said, for breaking up private or back-scoped methods, I think it would be simpler for me to push this and for you to send a new pull.